qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -520,11 +535,21 @@ def from_circuit(self, circuit):
         append_circuit(elem, circuit)
         return elem
 
-    def compose(self, other):
+    def to_matrix(self):
+        """Convert operator to Numpy matrix."""
+        return self.to_operator().data
+
+    def to_operator(self):
```

Comment:
The `decompose_cnotdihedral` function this eventually calls has an `assert` in it that should be replaced with an exception. Something like `if n > 2: raise NotImplementedError('Cannot decompose for N>2 qubits.')`

Reply:
Added `decompose_cnotdihedral_general` function that handles any number of qubits.